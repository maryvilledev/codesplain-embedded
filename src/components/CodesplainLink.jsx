import React from 'react';

const CodesplainLink = ({ snippetKey, style}) => {
  const properAddress = `https://www.codesplain.io/${snippetKey}`;
  return (
    <a href={properAddress}><h4 style={style}>{snippetKey}</h4></a>
  );
}

export default CodesplainLink;
