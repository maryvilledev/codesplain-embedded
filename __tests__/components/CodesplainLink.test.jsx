import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json'

import CodesplainLink from '../../src/components/CodesplainLink';

describe('<CodesplainLink />', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(<CodesplainLink snippetKey="birdperson.py"/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
