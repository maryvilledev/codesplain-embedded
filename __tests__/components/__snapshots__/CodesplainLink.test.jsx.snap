// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CodesplainLink /> matches snapshot 1`] = `
<a
  href="https://www.codesplain.io/birdperson.py"
  rel="noopener noreferrer"
  target="_blank"
>
  <h4>
    birdperson.py
  </h4>
</a>
`;
